kqueen/auth/ldap.py

```diff
@@ -61,30 +92,42 @@ def verify(self, user, password):
             None: Invalid credentials, Authentication failure.
             User: authenticated user.
         """
+        matched_dn = self._get_matched_dn(user.username)
+
+        for dn in matched_dn:
+            if self._bind(dn, password):
+                full_dn = dn
+
+        if full_dn:
+            logger.debug('Valid full-DN found: {}. It will be stored in user metadata'.format(full_dn))
+            return user, None
+
+        return None, "Verification failed"
 
-        dn = self._email_to_dn(user.username)
+    def _bind(self, dn, password):
 
         try:
+            self.connection = ldap.initialize(self.uri)
```

Comment:
already initialized in the constructor

Reply:
no, previously we use unbind() only once, and all works fine
but in this case we use unbind multiple times, and each time we should re-init ldap object, otherwise if=t failed to bind. i can clarify info directly or take a look on LDAP docs